### DISCUSSION TITLE
OpenSearch via HTTPS (encrypted connection)

### DISCUSSION THREAD

**Discussion by manuviens** [mastodon/mastodon#27463] [Q&A] [answered]
I want to host OpenSearch on a different server from Mastodon. If I configure the opensearch-security module to accept only HTTPS connections (OpenSearch will use the Let's Encrypt certificate) in addition to user/pass authentication, will Mastodon be able to connect to it? In the documentation (https://docs.joinmastodon.org/admin/elasticsearch/) there doesn't seem to be any mention of an encrypted connection, no variable is mentioned for that.

**Comment by mszucs** [comment, marked as answer]
```
ES_HOST=https://your.opensearch.host
ES_PORT=<port>
ES_USER=<user>
ES_PASS=<pass>
```
works.

For us, this configuration triggered a bug that caused Sidekiq to accumulate open connections to Elasticsearch.

This was fixed with [41f8110](https://github.com/nilsding/mastodon/commit/41f8110c114be1ae91e60991cb5f882733064e62).
